tutorials/transparent-proxy-and-filtering-on-k8s-with-initializers/index.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Transparent Proxy and Filtering on Kubernetes with Initializers
+description: Learn how to run a transparent proxy on Kubernetes to filter and intercept traffic out of your deployments using custom deployment initializers.
+author: danisla
+tags: Kubernetes mitmproxy proxy
+date_published: 2017-09-23
+---
+
+Dan Isla | Google Cloud Solution Architect | Google
+
+This is a follow-on tutorial to the [Transparent Proxy and Filtering on K8S](https://cloud.google.com/community/tutorials/transparent-proxy-and-filtering-on-k8s) tutorial. It shows how to simplify the application of a transparent proxy for existing deployments using a Deployment Initializer. Initializers are one of the [Dynamic Admission Control](https://kubernetes.io/docs/admin/extensible-admission-controllers/) features of Kubernetes, and are available as an alpha feature in Kubernetes 1.7.
+
+This tutorial uses the [tproxy-initializer](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-initializer) Kubernetes Initializer to inject the sidecar InitContainer, ConfigMap and environment variables into a deployment when the annotation `"initializer.kubernetes.io/tproxy": "true"` is present. This tutorial also demonstrates how to deploy the [tproxy Helm chart](https://github.com/danisla/kubernetes-tproxy/tree/master/charts/tproxy) with the optional [Role Based Access Control](https://kubernetes.io/docs/admin/authorization/rbac/) support.
+
+Just like in the previous tutorial, the purpose of the [tproxy-sidecar](https://github.com/danisla/kubernetes-tproxy/tree/master/sidecar) container is to create firewall rules in the pod network to block egress traffic. The [tproxy-podwatch](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-podwatch) controller watches for pod changes containing the annotation and automatically add/removes the local firewall `REDIRECT` rules to apply the transparent proxy to the pod.
+
+![architecture diagram](./tproxy_initializers_diagram.png)
+
+**Figure 1.** transparent proxy with initializers architecture diagram
+
+## Objectives
+
+- Create a Kubernetes cluster with initializer and RBAC support using Google Container Engine
+- Deploy the tproxy, tproxy-initializer and the tproxy-podwatch pods using Helm
+- Deploy example apps with annotations to test external access to a Google Cloud Storage bucket
+
+## Before you begin
+
+This tutorial assumes you already have a Google Cloud Platform (GCP) account and are familiar with the high level concepts of Kubernetes Pods and Deployments.
+
+## Costs
+
+This tutorial uses billable components of GCP, including:
+
+- [Google Container Engine](https://cloud.google.com/container-engine/pricing)
+
+Use the [Pricing Calculator](https://cloud.google.com/products/calculator/#id=f52c2651-4b02-4da3-b8cd-fdbca6ad89a9) to estimate the costs for your environment. The resources used in this tutorial are approximately $2.39 per day.
```

Comment:
don't include the price as it can easily change.

Reply:
I agree that it can change but I'll defer to @ccarpentiere or @jmdobry on what we should do here because I've had tech writers ask me to add the price when it wasn't there.

Reply:
The link to the calculator is sufficient

Reply:
Ok. I remove the price and left the link.